src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -278,7 +278,7 @@ public int login(String gatewayUrl, ShardInfo shardInfo, Compression compression
 
         String token = authConfig.getToken();
         setStatus(Status.LOGGING_IN);
-        if (token == null || token.isEmpty())
+        if (token.isEmpty())
```

Comment:
THIS WILL NPE.

Reply:
Check this class. Nothing like that will happen. 

https://github.com/DV8FromTheWorld/JDA/blob/development/src/main/java/net/dv8tion/jda/internal/utils/config/AuthorizationConfig.java